stix2/base.py

```diff
@@ -124,11 +124,20 @@ def __len__(self):
 
     # Handle attribute access just like key access
     def __getattr__(self, name):
-        return self.get(name)
+        # If the requested attribute is not None return its value
```

Comment:
This code still doesn't achieve consistency between dict and attribute access.  This will say that any *legal* attribute exists (for attribute access), but only *set* keys exist (for dict access).

Greg's suggestion would work (can even just do `self[name]`).  Or just delete the `elif name in self._properties` part.  (Checking `self.get(name)` against `None` technically can't distinguish between the key not existing, and the key existing with value `None`.  But with `None` used for "un-setting" a property, I guess that can't happen.)